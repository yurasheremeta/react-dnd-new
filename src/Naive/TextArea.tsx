import * as React from 'react';

interface TextAreaProps {
    value: string
    onChange: any
}
class TextArea extends React.Component<TextAreaProps>{

    render(){
        
        return( 
            <textarea value={this.props.value} onChange={this.props.onChange}/>
        )
    }
}

export default TextArea;