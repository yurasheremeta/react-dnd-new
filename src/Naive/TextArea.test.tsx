import * as React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import TextArea from './TextArea';
import 'jsdom-global/register';

describe('TextArea tests', () => {

    it('Render correctly TextArea' , () => {
        const wrapper = renderer.create(<TextArea value={"aaaaa"} onChange={() => {}}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    })

    it('Render right value' , () => {
        const defaultProps = {
            value: "Hello",
            onChange: () => {}
        }
        const wrapper = mount(<TextArea {...defaultProps} />);
        expect(wrapper.prop('value')).toEqual("Hello");
    })
})

