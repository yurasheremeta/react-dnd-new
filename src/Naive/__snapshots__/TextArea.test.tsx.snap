// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea tests Render correctly TextArea 1`] = `
<textarea
  onChange={[Function]}
  value="aaaaa"
/>
`;
